core/src/com/unciv/UncivGame.kt

```diff
@@ -294,7 +294,7 @@ class UncivGame(parameters: UncivGameParameters) : Game() {
                 restoreDefault = { musicController.resume() },
                 action = {
                     Gdx.app.exit()
-
+                    System.exit(0)
```

Comment:
Android Studio suggests to replace this line with `exitProcess(0)`.
P.S> I remember I have spent a good potion of time to overcome this bug, but not remember why I've given up.

Reply:
I have a new plan. call `System.exit(0)` at the end of `dispose()`.